### ISSUE TITLE
Custom types

### ISSUE THREAD

**Issue by wavebeem** [wavebeem/pkmn.help#159]
Being able to enter a custom type chart would be cool for Fakemon games

**Comment by wavebeem** [comment]
## Difficulties

- Translations... I guess custom types just won't be translated

- Colors... What color should the new types be?

- Enum... the types are an enumeration right now, but would need to be an arbitrary string type if we want to allow custom types

- Import format... Should it be a CSV upload again?

  - If not, how much work do we put in to designing a form to create a type matchup table

    - And should it even display like a table? Most of the entries are blank, so it might be easier to design the form differently

  - How do you share your custom types with other people?

    - I don't want to to approve and import people's custom fan game type lists centrally into pkmn.help, so there needs to be an easier way to share this info...

    - It could maybe be added to the query params as a custom parameter so you can directly link people?

  - pkmn.help should probably store this custom type table so you don't have to constantly import it again

**Comment by gingerchew** [comment]
For colors, there is a way to make your Pokemon "type-less" using Incinerate with a Fire type Pokemon. After using Incinerate, the Pokemon loses it's Fire typing, so Blazekin (Fire/Fighting) becomes a Fighting type. A Pokemon that is *only* Fire type, like Arcanine, would end up without a type. 

On Pokemone Showdown, they resolve this using this badge for it's type: ![](https://play.pokemonshowdown.com/sprites/types/%3F%3F%3F.png)

The background is (according to my color picker, so results vary) #6fa08f. That could make for a good starting point for a custom type color.

**Comment by wavebeem** [comment]
@gingerchew wow... is a type-less Pokémon hit neutrally (1x) by every move? that's so weird haha

**Comment by gingerchew** [comment]
I don't even know, I've only seen it in an advanced tactic that transfers the Wonder Guard ability from a Shedinja to an opponent then onto the Arcanine once it is typeless. Couldn't even begin to tell you how that set up is possible.

It's gotta be 1x though, might be worth it to demo it out on Pokémon Showdown and see to be sure?

**Comment by gingerchew** [comment]
Here's a showdown replay of someone using that tactic: https://replay.pokemonshowdown.com/gen7doublesoususpecttest-651575368

**Comment by wavebeem** [comment]
Just got another visitor who requested this feature.

**Comment by wavebeem** [comment]
Just got another request for this feature

**Comment by wavebeem** [comment]
Another request for this feature

**Comment by wavebeem** [comment]
Another request for this feature

**Comment by RobinDeFloof** [comment]
Some suggestions but before I say anything I'm prefacing this with I have no idea what I'm yapping about j barely understand how coding works lmfao
I'm like the person who designs the models and schematics of builds while the people in suggesting things to (in this case you) are the ones who do the actual work lmfao /silly

But back to the point:
I'd say have a custom page for making, editing and maybe importing and exporting what I'm gonna call "type packs"

Translations would probably be mostly on user end, so making edits to downloaded "type packs" and whatnot, alternatively have the option for the maker to add alternative names for different languages

Someone already mentioned the old "???" Type color and I'd say that's a good default for when you make the new type, along with either a hex code I put and/or color wheel to select the color 

As for sharing one idea that comes to mind is just a discord, poke community, and/or a reddit page for sharing these, and when imported the data can be saved to your device for easy access

(Image for reference on how I'd imagine it looking in my head tbh lol)

<img width="1890" height="954" alt="Image" src="https://github.com/user-attachments/assets/c832a8a1-f5c3-44c2-83ea-bb3f8ff5116f" />

**Comment by wavebeem** [comment]
That's an interesting mockup! Thanks for sharing your idea.

I've definitely seen a lot of people mentioning the possibility of modifying built-in types too, like "what if bug was better" and such.

Part of me wants to offload this to a spreadsheet, but I also personally can't stand looking at 18x18 grids of type matchups

The nice thing about offloading it to a spreadsheet though is it makes it more obviously shareable, as you can just make a public spreadsheet on google docs, or pin a CSV in a discord server, or whatever

i _will not_ get into the realm of hosting user generated content (legal minefield/expensive/lot of effort), so there would need to be some way to share this stuff

please let me know if you get around to trying the custom pokedex feature, as i'd like to model "custom type charts" the same way... but i just dont have a lot of feedback there about if people like how that feature works

**Comment by wavebeem** [comment]
<img width="906" height="227" alt="Image" src="https://github.com/user-attachments/assets/b7f5495b-6faf-44e0-b2b5-0e9a87ee413c" />

something like this but extended to all 18+ types honestly doesn't seem that bad, and is an easily shareable file in plain text thru any chat program and takes little space... its not PRETTTY, but users can customize the spreadsheet in google docs and share templates from there if they prefer

**Comment by wavebeem** [comment]
oh yeah, I hate looking at tables and grids. that's one of the reasons I made the site lol. i only begrudgingly used a table for team defense mode results because everything else I could think of felt worse (and I still think about redesigning it sometimes lol)

**Comment by wavebeem** [comment]
good point. the simplest slice here i can think of is "import/export CSV type table with colors" and just display an error on the pokedex screen saying its not supported with custom types

still no promises about when i'll get around to it, but i think "limited feature" is better than "perfect feature that never exists" lol

**Comment by RobinDeFloof** [comment]
True! Little steps first lol

**Comment by wavebeem** [comment]
Another request for this.

**Comment by wavebeem** [comment]
Another one!
